src/MongoDB.Driver.Core/Core/Servers/ServerMonitor.cs

```diff
@@ -319,7 +338,10 @@ private void SetDescription(ServerDescription newDescription)
         private void SetDescription(ServerDescription oldDescription, ServerDescription newDescription)
         {
             Interlocked.Exchange(ref _currentDescription, newDescription);
-            OnDescriptionChanged(oldDescription, newDescription);
+            if (ShouldDescriptionChangedEventBePublished(oldDescription, newDescription))
```

Comment:
The specification states:

> Server description for each server MUST be updated (replaced) every heartbeat. However, new description MUST NOT cause the SDAM flow to be executed if the new description is equal, as defined in this section, to the previous description.

I'm not positive, but this change looks like it trades one bug for another.  With this change, the SDAM flow will, correctly, not be executed (and no events will be published either), but it will also fail to update the server description within the cluster description.  It's important that this update occurs, so that server selection sees the latest roundTripTime value (which is not included in the equality comparison).

It's definitely possible I'm misunderstanding the code, but this is what I suspect.

Reply:
thank you for clarification. In the current version I left only logic to supress `ServerDescriptionChangedEvent` and `ClusterDescriptionChangedEvent` events